var/httpd/htdocs/js/Core.Agent.Admin.ProcessManagement.js

```diff
@@ -1716,6 +1719,73 @@ Core.Agent.Admin.ProcessManagement = (function (TargetNS) {
+                // alert if there is already an input
+                FieldsWithValue = $('#ConfigParams input[name*="ConfigValue"]').filter(function() { return $(this).val() !== ''; }).length;
+                if (FieldsWithValue) {
+                    Core.UI.Dialog.ShowDialog({
+                        Title : Core.Language.Translate('Warning'),
+                        HTML : Core.Language.Translate('Are you sure you want to overwrite the config parameters?'),
+                        Modal : true,
+                        CloseOnClickOutside : true,
+                        CloseOnEscape : true,
+                        PositionTop: '100px',
+                        PositionLeft: 'Center',
+                        Buttons: [
+                            {
+                                Label: Core.Language.Translate("Yes"),
+                                Class: 'Primary',
+                                Function: function () {
+                                    TargetNS.GetDefaultConfigParameters(TransitionActionModule)
+                                    Core.UI.Dialog.CloseDialog($('.Dialog:visible'));
+                                }
+                            },
+                            {
+                                Label: Core.Language.Translate("Cancel"),
+                                Function: function () {
+                                    Core.UI.Dialog.CloseDialog($('.Dialog:visible'));
+                                }
+                            }
+                        ]
+                    });
+                } else {
+                    TargetNS.GetDefaultConfigParameters(TransitionActionModule);
+                }
+            }
+        });
+    }
+    
+    TargetNS.GetDefaultConfigParameters = function (TransitionActionModule) {
+
+        // do AJAX call
+        var Data = {
+            Action: 'AdminProcessManagementTransitionAction',
+            Subaction: 'GetDefaultConfigParameters',
+            Module: TransitionActionModule
+        };
+        Core.AJAX.FunctionCall(Core.Config.Get('CGIHandle'), Data, function (Response) {
+            if (Response) {
+                // delete old params
+                $('#ConfigParams > fieldset').remove();
+                // add one empty element
+                if (!Object.keys(Response).length) {
+                    $('#ConfigParams').append( $('#ConfigParamContainer').html().replace(/_INDEX_/g, 1) );
+                }
+                // insert params from config
+                $.each(Object.keys(Response).sort(), function(Counter, Key) {
+                    var $Element = $($('#ConfigParamContainer').html().replace(/_INDEX_/g, Counter + 1));
+                    $Element.find('input[name*="ConfigKey"]').val(Key);
+                    $Element.find('input[name*="ConfigValue"]').attr('placeholder', Response[Key]);
+                    $('#ConfigParams').append($Element);
+                });
+            }
+        });
+    }
+
```

Comment:
```suggestion
    TargetNS.InitDefaultConfigParameters = function () {
        $('#Module').on("change", function () {
            var Module = $("#Module option:selected").val()

            if (!Module) {
                return;
            }

            Core.UI.Dialog.ShowDialog({
                Title:               Core.Language.Translate('Warning'),
                HTML:                Core.Language.Translate('Are you sure you want to overwrite the config parameters?'),
                Modal:               true,
                CloseOnClickOutside: true,
                CloseOnEscape:       true,
                PositionTop:         '100px',
                PositionLeft:        'Center',
                Buttons: [
                    {
                        Label: Core.Language.Translate("Yes"),
                        Class: 'Primary',
                        Function: function () {
                            var ConfigParameters = TargetNS.GetDefaultConfigParameters(Module)
                            TargetNS.SetDefaultConfigParameters(ConfigParameters)
                            Core.UI.Dialog.CloseDialog($('.Dialog:visible'));
                        }
                    },
                    {
                        Label: Core.Language.Translate("Cancel"),
                        Function: function () {
                            Core.UI.Dialog.CloseDialog($('.Dialog:visible'));
                        }
                    }
                ]
            });
        });
    }

    TargetNS.GetDefaultConfigParameters = function (Module) {

        // do AJAX call
        var ConfigParameters = {},
            Data = {
            Action:    'AdminProcessManagementTransitionAction',
            Subaction: 'GetDefaultConfigParameters',
            Module:    Module,
        };

        Core.AJAX.FunctionCallSynchronous(Core.Config.Get('CGIHandle'), Data, function (Response) {
            ConfigParameters =  Response;
        });

        return ConfigParameters;
    }

    TargetNS.SetDefaultConfigParameters = function (ConfigParameters) {

        if (!ConfigParameters){
            return;
        }
        // delete old params
        $('#ConfigParams > fieldset').remove();

        // add one empty element
        if (!Object.keys(ConfigParameters).length) {
            $('#ConfigParams').append( $('#ConfigParamContainer').html().replace(/_INDEX_/g, 1) );
        }
        // insert params from config
        $.each(Object.keys(ConfigParameters).sort(), function(Counter, Key) {
            var $Element = $($('#ConfigParamContainer').html().replace(/_INDEX_/g, Counter + 1));
            $Element.find('input[name*="ConfigKey"]').val(Key);
            $Element.find('input[name*="ConfigValue"]').attr('placeholder', ConfigParameters[Key]);
            $('#ConfigParams').append($Element);
        });
    }
    
```

Reply:
You found some great improvements for the JS module. Thank you. But you removed the variable FieldsWithValue which checks, if the user already typed something into the form. Without this condition the popup appears everytime the user chooses a module. I just checked the code on my develop system. The popup appears also after the first selection. I think this is a bit confusing.

Reply:
With the code of your suggestion, the popup appears always after choosing a module, also after the first pick.
I would prefer a check, if there is any data in the form or if the user chooses the module the first time. Otherwise the message "Are you sure you want to overwrite the config parameters?" will be misunderstand if there are no config parameters in the form.

Reply:
Ok, you mean:

If there is no data in the form => no message
If data exists => message

Thats sound good.